pages/tutorials/[slug].js

```diff
@@ -30,10 +33,70 @@ export async function getStaticProps(context) {
 }
 
 export default function Tutorial({ name, steps, id, slug }) {
+  const [currentStep, setCurrentStep] = useState(0);
+
   return (
     <>
       <Headline>{name}</Headline>
-      <TutorialStartCard />
+      {currentStep === 0 ? (
+        <>
+          <TutorialStartCard />
+          <ButtonContainer>
+            <Button
+              isPrimary={true}
+              onClick={() =>
+                setCurrentStep((prevCurrentStep) => prevCurrentStep + 1)
+              }
+            >
+              start
+            </Button>
+          </ButtonContainer>
+        </>
+      ) : currentStep < steps.length ? (
+        <>
+          <TutorialCard step={steps[currentStep]} />
+          <ButtonContainer>
+            <Button
+              isPrimary={false}
+              onClick={() =>
+                setCurrentStep((prevCurrentStep) =>
+                  prevCurrentStep > 0 ? prevCurrentStep - 1 : 0
+                )
+              }
+            >
+              prev
+            </Button>
+            <Button
+              isPrimary={true}
+              onClick={() =>
+                setCurrentStep((prevCurrentStep) =>
+                  prevCurrentStep < steps.length
+                    ? prevCurrentStep + 1
+                    : prevCurrentStep
+                )
+              }
+            >
+              next
+            </Button>
+          </ButtonContainer>
+        </>
+      ) : (
+        <>
+          <h2>done</h2>
+          <ButtonContainer>
+            <Button
+              isPrimary={false}
+              onClick={() =>
+                setCurrentStep((prevCurrentStep) =>
+                  prevCurrentStep > 0 ? prevCurrentStep - 1 : 0
```

Comment:
i can see the same function in l 62/63 -> maybe define it some place else and use it here.

Reply:
Thanks, Mark!
I thought that only one repetition of this line would not attract attention, but of course I did not think of you having such a keen eye. 😄
I refactored the onclick for adding a step as well.